src/muse/wizard.py

```diff
@@ -0,0 +1,244 @@
+import os
+from pathlib import Path
+from typing import Callable, List
+
+import pandas as pd
+from tomlkit import dumps, parse
+
+
+def modify_toml(path_to_toml: Path, function: Callable):
+    """Apply the specified function to modify a toml file.
+
+    Args:
+        path_to_toml: Path to the toml file
+        function: Function to apply to the toml data. Must take a dictionary as a single
+            input and modify it in place.
+
+    """
+    data = parse(path_to_toml.read_text())
+    function(data)
+    path_to_toml.write_text(dumps(data))
+
+
+def get_sectors(model_path: Path) -> List[str]:
+    """Get a list of sector names for a model.
+
+    Args:
+        model_path: Path to the model folder.
+
+    Returns:
+        List of sector names
+    """
+    return [
+        s.name
+        for s in (model_path / "technodata").iterdir()
+        if (s / "Technodata.csv").is_file()
+    ]
+
+
+def add_new_commodity(
+    model_path: Path, commodity_name: str, sector: str, copy_from: str
+) -> None:
+    """Add a new commodity to a sector by copying an existing one.
+
+    Args:
+        model_path: Path to the model folder.
+        commodity_name: Name of the new commodity.
+        sector: Sector to add the commodity to.
+        copy_from: Name of the commodity to copy from.
+    """
+    # Add commodity to global commodities file
+    global_commodities_file = model_path / "input/GlobalCommodities.csv"
+    df = pd.read_csv(global_commodities_file)
+    new_rows = df[df["Commodity"] == copy_from.capitalize()].assign(
+        Commodity=commodity_name.capitalize(), CommodityName=commodity_name
+    )
+    df = pd.concat([df, new_rows])
+    df.to_csv(global_commodities_file, index=False)
+
+    # Add columns to additional files
+    files_to_update = [
+        model_path / file
+        for file in [
+            f"technodata/{sector}/CommIn.csv",
+            f"technodata/{sector}/CommOut.csv",
+            "input/BaseYearImport.csv",
+            "input/BaseYearExport.csv",
+            "input/Projections.csv",
+        ]
+    ] + list((model_path / "technodata/preset").glob("*"))
+
+    for file in files_to_update:
+        df = pd.read_csv(file)
+        df[commodity_name] = df[copy_from]
+        df.to_csv(file, index=False)
+
+
+def add_new_process(
+    model_path: Path, process_name: str, sector: str, copy_from: str
+) -> None:
+    """Add a new process to a sector by copying an existing one.
+
+    Args:
+        model_path: Path to the model folder.
+        process_name: Name of the new process.
+        sector: Sector to add the process to.
+        copy_from: Name of the process to copy from.
+    """
+    files_to_update = [
+        model_path / file
+        for file in [
+            f"technodata/{sector}/CommIn.csv",
+            f"technodata/{sector}/CommOut.csv",
+            f"technodata/{sector}/ExistingCapacity.csv",
+            f"technodata/{sector}/Technodata.csv",
+        ]
+    ]
```

Comment:
If you use round brackets you can leave this as a generator expression and then Python doesn't have to create a list object:

```suggestion
    files_to_update = (
        model_path / file
        for file in (
            f"technodata/{sector}/CommIn.csv",
            f"technodata/{sector}/CommOut.csv",
            f"technodata/{sector}/ExistingCapacity.csv",
            f"technodata/{sector}/Technodata.csv",
        )
    )
```

Again, just a nit, so not important